### PR TITLE
THE RUINCUT [AFFRONT TO GOD]

### PR DESCRIPTION
**Pull request by Zevotech**
## About The Pull Request
Okay, so heres the gist. **We have a lot of really bad ruins that don't meet the current standards.** They might have been fine by older standards, but for a long time we maptainers have needed to put our foot down and pick what stays and what goes. 
This PR does that. And a lot of other stuff.

This PR:
Removes a massive amount of map files for ruins, leaving most naturally-generating planets with only *4-7 unique ruins* each. 
Removes the areas and code for those ruins, as well as a large amount of unused or already deleted ruins that were just rotting in our codebase.
Fixes the issue where sometimes a planet would generate without a ruin by removing ruin datums that lacked an associated map file.
Makes adjustments to a few of the remaining ruins for the purpose of balance. Anything that needed minor changes to pass standards was changed.

<details open>
<summary>List of changes:</summary>
<br>
Ruin Deletions:

beach_colony
beach_crashed_starwalker (remapped onto lavaland)
beach_knights_rest
icemoon_surface_corporate_rejects
icemoon_surface_engioutpost
icemoon_underground_abandoned_newcops
icemoon_underground_drakelair
jungle_botany
jungle_nest
jungle_pirate
jungle_pizzawave
jungle_seedling
jungle_surface_bombmakers_cabin
jungle_surface_coffinpirate
jungle_surface_ikea_ai
jungle_surface_ninjashrine
jungle_surface_roomates
jungle_surface_weed_shack
lavaland_surface_SPOOKYCRASH
lavaland_surface_codelab
lavaland_surface_comm_outpost
lavaland_surface_dwarffortress
lavaland_surface_heirophant
lavaland_surface_pizzaparty
lavaland_surface_random_ripley
lavaland_surface_sloth
lavaland_surface_survivalpod
lavaland_surface_xeno_nest
lavaland_surface_crashed_pinnance
rockplanet_boxsci
rockplanet_crash
rockplanet_dangerpod
rockplanet_heirophant
rockplanet_house
rockplanet_miningexpedition
rockplanet_pioneer
rockplanet_saloon
whitesands_surface_assaultpodcrash
whitesands_surface_camp_drugstore
whitesands_surface_chokepoint
whitesands_surface_conveniencestore
whitesands_surface_onlyaspoonful
whitesands_surface_starfurycrash
whitesands_surface_waterplant
Fast_Food
crashedship
cryocontainment
dangerous_research
dark_glade
excavator_DK
fueldepot
gondolaasteroid
hellfactory
ntfacility
oretruck
provinggrounds
scav_mining
transport18
vaporwave
wasteplanet_clowncrash
wasteplanet_crash_kitchen
wasteplanet_fortress_of_solitude
wasteplanet_ore_proccessing_facility
wasteplanet_pod
wasteplanet_tarpit
wasteplanet_tradepost

Code File Deletions:
lavaland_ruin_code.dm
biodome_clown_planet.dm
codelab.dm
pizzaparty.dm
sloth.dm
asteroid4.dm
caravanambush.dm
clericsden.dm
crashedclownship.dm
crashedship.dm
deepstorage.dm
forgottenship.dm
hellfactory.dm
hilbertsholtel.dm
listeningstation.dm
spacehotel.dm

Changes:
beach_fishing_hut (replaced pirates with frontiersmen)
beach_pirate_crash (replaced pirates with frontiersmen)
beach_treasure_cove (replaced pirates with frontiersmen)
icemoon_underground_abandoned_village (fixed a couch)
jungle_syndicate (removed maid outfits)
lavaland_surface_lava_canyon (removed cleric mace)
whitesands_surface_camp_saloon (fixed an item with a missing icon state)
whitesands_surface_medipen_plant (rearranged some of the loot, removed mining, booze and med vendor)
astraeus (fixed incorrect rock type)
singularity_lab (fixed broken icons on exterior rock walls, removed a lootdrop from a removed ruin, cut down on length slightly to prevent clipping into z-level walls)
spacemall (removed pamphlet that was from a removed ruin)
rockplanet_harmfactory (fixed keycard doors that were reused from a deleted ruin)
wasteplanet_lab (removed free mining vendor)
independent_beluga (removed pamphlet that was from a removed ruin)

Also removes a lot of items/code from deprecated/removed ruins from various other files
</details>

## What Now?
The maptainer team is working on compiling guides and information regarding map contribution in a forum on the discord. In that forum, there will be posts with guides on how to contribute, how to map, and how to make good ruins. We're trying to streamline the process, and make it easier for people to scale the learning curve and make good maps.
### We need YOU, testers, to read those posts and help us to make the best damn ruins this server has ever seen.
Here are the ruin amounts post-cut that remain:
Lavaplanet: 4
Jungleplanet: 7
Beachplanet: 7
Iceplanet: 4
Sandplanet: 4
Wasteplanet: 4
Rockplanet: 4
Space: 7

## Why It's Good For The Game
If you've ever seen the sloth ruin on lavaplanet, then you knew this PR was a long time coming. A lot of the maps that have been removed were explicitly bad enough for us to not want players to use them as examples, as well as pruning small ruins and fixing others.

The main goal is for our lack of good ruins to be visible, without all of the shitty ones getting in the way of how much we need new ones. Good ruins aren't hard to make, it's just that our standards and priorities have evolved to where a lot of ruins that USED to be good are now really, really bad.
## Changelog

:cl:
fix: Planets should no longer spawn without ruins when generating due to bad map datums
del: Removed 3 Beach ruins
del: Removed 4 Iceplanet ruins
del: Removed 11 Jungle ruins
del: Removed 10 Lavaland ruins
del: Removed 8 Rockplanet ruins
del: Removed 7 Sandplanet ruins
del: Removed 7 Rockplanet ruins
del: Removed 15 Space ruins
del: Removed 16 unused/removed ruin code files
del: Removed a whole lot of old/unused ruin code from misc files
tweak: Tweaked various ruins and one ship to fix issues and remove unused items
/:cl:

### CODE REVIEW

**Comment by Zevotech** [comment]
Against all known laws of physics, god, and nature, this passes checks.

**Comment by rye-rice** [comment]
still on target for the 10 minimum rockplanet and wasteplanet ruins for overmap 4 to be merged! wooo!

**Comment by Zevotech** [comment]
i'll make up for it after this is merged.

File: code/modules/mob/living/simple_animal/constructs.dm [hunk 1]
```diff
@@ -408,6 +408,29 @@
 	seek.Grant(src)
 	seek.Activate()
 
+///////////////////////PROTEONS (from deprecated clericden ruin)//////////////////////
+
+/mob/living/simple_animal/hostile/construct/proteon
```
Comment (goober3) [inline code/modules/mob/living/simple_animal/constructs.dm]: delete this. we don't want cult constructs anyway.

**Comment by Zevotech** [comment]
should be ready for TM once checks pass

**Pull request merged**